### PR TITLE
shorten scheduling priorities taint key

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it: shortens the taint key to be of a valid length.

See: https://github.com/kubernetes/kubernetes/pull/97819/files#r575078085

TODO: might need to update the kubectl docs.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/scheduling/priorities.go [hunk 1]
```diff
@@ -637,7 +637,7 @@ func createRC(ns, rsName string, replicas int32, rcPodLabels map[string]string,
 
 func getRandomTaint() v1.Taint {
 	return v1.Taint{
-		Key:    fmt.Sprintf("kubernetes.io/scheduling-priorities-e2e-taint-key-%s", string(uuid.NewUUID())),
+		Key:    fmt.Sprintf("kubernetes.io/e2e-scheduling-priorities-%s", string(uuid.NewUUID()[:23])),
```
Comment (BenTheElder) [inline test/e2e/scheduling/priorities.go:640]: IMO we should prefer keys you can do a simple prefix match on, but admittedly moving this to the front was not important. "-taint-key" seems redundant. This should now be 40 + 23 characters.

I don't think having a valid UUID is important, and we should have enough bits, I think.

**Comment by aojea** [comment]
we start to have many flakes in kind again lately, storage ... last time were resources problem causing this IIRC

**Comment by BenTheElder** [comment]
there's a couple of other things causing CI failures right now across all jobs so I've been a little more focused on that, but I have noticed the e2e as well. (see e.g. https://github.com/kubernetes/test-infra/issues/20816)

**Comment by BenTheElder** [comment]
Apparently the kind jobs are only requesting 4 cpu which is less than verify ... I don't remember doing this.

**Comment by gautamdevkhanna** [comment]
Unsubscribe 

Warm Regards
Gautamdev
09930928460

**Comment by BenTheElder** [comment]
integration seems super flaky now
/retest

**Review by damemi — COMMENTED** [review]
@BenTheElder thanks for fixing this! Sorry I was missing the failed test runs...

**Comment by BenTheElder** [comment]
No worries :-)

**Pull request merged**